lib/sycamore/sycamore/evaluation/evaluate.py

```diff
@@ -0,0 +1,137 @@
+from typing import Dict, List, Union
+from abc import ABC, abstractmethod
+
+import os
+from sycamore.data import Element, Document
+from sycamore.evaluation.pipeline import EvaluationPipeline
+from sycamore.evaluation import EvaluationDataPoint
+from sycamore.evaluation.metrics import document_retrieval_metrics, rouge_metrics
+from sycamore import Context
+
+
+def add_searchContext_to_datapoint(datapoint: Document) -> List[Element]:
+    assert isinstance(datapoint, Document)
+    source_documents: List[Element] = []
+    assert datapoint.get("raw") is not None
+    for search_result in datapoint["raw"].get("SearchContexts", []):
+        source_document = Element()
+        properties = {
+            "_location": search_result["document_url"],
+            "page_number": search_result["page_numbers"][0],
+            "doc_id": search_result["document_id"],
```

Comment:
should I worry about "document_url", "page_numbers", and "document_id" not being present?

Reply:
missing value shouldn't cause problem for url and doc_id